lib/plugins/aws/deploy/lib/create-stack.js

```diff
@@ -70,50 +69,49 @@ module.exports = {
+        StackName: stackName,
+      });
+
+      const shouldCheckStackOutput =
+        // check stack output only if acceleration is requested
+        this.provider.isS3TransferAccelerationEnabled() &&
+        // custom deployment bucket won't generate any output (no check)
+        !this.serverless.service.provider.deploymentBucket;
 
-              if (!isAlreadyAccelerated) {
-                log.info('Not using S3 Transfer Acceleration (1st deploy)');
-                this.provider.disableTransferAccelerationForCurrentDeploy();
-              }
-            }
+      if (shouldCheckStackOutput) {
+        const isAlreadyAccelerated = data.Stacks[0].Outputs.some(
+          (output) => output.OutputKey === 'ServerlessDeploymentBucketAccelerated'
+        );
 
-            const stackStatus = data.Stacks[0].StackStatus;
-            if (inactiveStateNames.has(stackStatus)) {
-              const errorMessage = [
-                'Service cannot be deployed as the CloudFormation stack ',
-                `is in the '${stackStatus}' state. `,
-                'This may signal either that stack is currently deployed by a different entity, ',
-                'or that the previous deployment failed and was left in an abnormal state, ',
-                "in which case you can mitigate the issue by running 'sls remove' command",
-              ].join('');
-              throw new ServerlessError(errorMessage, 'AWS_CLOUDFORMATION_INACTIVE_STACK');
-            }
-            return BbPromise.resolve('alreadyCreated');
-          })
-      )
-      .catch((e) => {
-        if (e.message.indexOf('does not exist') > -1) {
-          if (this.serverless.service.provider.deploymentBucket) {
-            this.createLater = true;
-            return BbPromise.resolve();
-          }
-          return BbPromise.bind(this).then(this.create);
+        if (!isAlreadyAccelerated) {
+          log.info('Not using S3 Transfer Acceleration (1st deploy)');
+          this.provider.disableTransferAccelerationForCurrentDeploy();
         }
-        throw e;
-      });
+      }
+
+      const stackStatus = data.Stacks[0].StackStatus;
+      if (inactiveStateNames.has(stackStatus)) {
+        const errorMessage = [
+          'Service cannot be deployed as the CloudFormation stack ',
+          `is in the '${stackStatus}' state. `,
+          'This may signal either that stack is currently deployed by a different entity, ',
+          'or that the previous deployment failed and was left in an abnormal state, ',
+          "in which case you can mitigate the issue by running 'sls remove' command",
+        ].join('');
+        throw new ServerlessError(errorMessage, 'AWS_CLOUDFORMATION_INACTIVE_STACK');
+      }
+      return 'alreadyCreated';
+    } catch (e) {
+      if (e.message.indexOf('does not exist') > -1) {
+        if (this.serverless.service.provider.deploymentBucket) {
+          this.createLater = true;
+          return Promise.resolve();
```

Comment:
Same here

Reply:
I am getting an eslint error if I were to return <empty>
https://eslint.org/docs/latest/rules/consistent-return